### PR TITLE
Nagios patterns

### PR DESCRIPTION
**Pull request by geogdog**
Originally logged as #23.  I've added tests and updated the pattern.

### CODE REVIEW

File: spec/patterns/nagios_spec.rb [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+  it "a pattern pass the grok expression" do
+    expect(grok).to pass
+  end
+
+  it "matches a simple message" do
+    expect(subject).to match(value)
+  end
+
+  it "generates the nagios_type field" do
+    expect(grok_match(subject, value)).to include("nagios_type" => "SERVICE ALERT")
```
Comment (purbon) [inline spec/patterns/nagios_spec.rb]: In here you can use the value created in line 113, no need to call again the pattern matching.

**Comment by purbon** [comment]
Hi, I must admit I'm not an expert on nagios log format. Can you also add test to check if this change still support the old format? this would be very nice.

**Comment by geogdog** [comment]
Hi,

As far I can tell...this format has always been incorrect.  Removing the semi-colon is actually a bug fix.  Maybe I could add a regression test to make sure it doesn't creep back in?

File: spec/patterns/nagios_spec.rb (hunk 1 above)
Comment (geogdog) [inline spec/patterns/nagios_spec.rb]: You mean replace the occurances of:
```
expect(grok_match(subject, value)).to
```
with just:
```
expect(grok).to
```
?

**Comment by purbon** [comment]
Hi,
  I did a search for this kind of logs and I agree this is a bug fix, looks like this has been always there, so now not sure why this got introduced. I think last things to get this merged are:

* Add a regression test.
* Do a minor version bump.
* Rebase the PR.

Looking forward to have this in.

**Comment by geogdog** [comment]
I've rebased against master and added the regression test for the TIMEPERIOD TRANSITION bug.

**Comment by purbon** [comment]
LGTM

**Pull request closed without merging**